src/lib/hooks/useBlockNumber.ts

```diff
@@ -1,32 +1,38 @@
 import useActiveWeb3React from 'hooks/useActiveWeb3React'
-import useDebounce from 'hooks/useDebounce'
 import useIsWindowVisible from 'hooks/useIsWindowVisible'
 import { atom } from 'jotai'
 import { useAtomValue, useUpdateAtom } from 'jotai/utils'
-import { useCallback, useEffect, useState } from 'react'
+import { useCallback, useEffect } from 'react'
 
-function useBlock() {
+interface ChainBlock {
+  chainId?: number
+  block?: number
+}
+const chainBlockAtom = atom<ChainBlock>({})
+
+function useUpdateChainBlock() {
   const { chainId, library } = useActiveWeb3React()
   const windowVisible = useIsWindowVisible()
-  const [state, setState] = useState<{ chainId?: number; block?: number }>({ chainId })
+  const setChainBlock = useUpdateAtom(chainBlockAtom)
 
   const onBlock = useCallback(
     (block: number) => {
-      setState((state) => {
-        if (state.chainId === chainId) {
-          if (typeof state.block !== 'number') return { chainId, block }
-          return { chainId, block: Math.max(block, state.block) }
+      setChainBlock((chainBlock) => {
+        if (chainBlock.chainId === chainId) {
+          if (chainBlock.block === block) return chainBlock
+          if (typeof chainBlock.block !== 'number') return { chainId, block }
+          return { chainId, block: Math.max(block, chainBlock.block) }
```

Comment:
Do you recall why the `Math.max` is used here? It feels like it could be wrong to me. Why would we be in a situation where `.max` determines correctness?

Reply:
my assumption is that a provider may rewind given its sources, and that this prevents that from being reflected in our ui, although this was copied over from interface code so idk for sure.